packages/typescript/src/experimental/transports/jsonRPC.ts

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { fromBase64 } from '@mysten/bcs';
+
+import { bcs } from '../../bcs/index.js';
+import type {
+	ObjectOwner,
+	SuiClient,
+	SuiObjectData,
+	SuiTransactionBlockResponse,
+} from '../../client/index.js';
+import { batch } from '../../transactions/plugins/utils.js';
+import { Transaction } from '../../transactions/Transaction.js';
+import { ObjectError } from '../errors.js';
+import type { Experimental_SuiClientTypes } from '../types.js';
+
+export class JSONRpcTransport implements Experimental_SuiClientTypes.TransportMethods {
+	#jsonRpcClient: SuiClient;
+
+	constructor(jsonRpcClient: SuiClient) {
+		this.#jsonRpcClient = jsonRpcClient;
+	}
+
+	async getObjects(options: Experimental_SuiClientTypes.GetObjectsOptions) {
+		const batches = batch(options.objectIds, 50);
+		const results: Experimental_SuiClientTypes.GetObjectsResponse['objects'] = [];
+
+		for (const batch of batches) {
+			const objects = await this.#jsonRpcClient.multiGetObjects({
+				ids: batch,
+				options: {
+					showOwner: true,
+					showType: true,
+				},
+			});
+
+			for (const [idx, object] of objects.entries()) {
+				if (object.error) {
+					results.push(ObjectError.fromResponse(object.error, batch[idx]));
+				} else {
+					results.push(parseObject(object.data!));
+				}
+			}
+		}
+
+		return {
+			objects: results,
+		};
+	}
+	async getOwnedObjects(options: Experimental_SuiClientTypes.GetOwnedObjectsOptions) {
```

Comment:
I'm fine with this but we did talk about the "is this actually the same thing as `getObjects` with a different filter". I think the pagination needs (bounded vs unbounded list) are _maybe_ a compelling reason to keep them separated, but with things like Multiparty objects I also think these bespoke methods will get trickier (we'll either have more of them, or we'll need to consolidate later)